src/test/ddms_tests.cpp

```diff
@@ -0,0 +1,233 @@
+#include <validation.h>
+#include <consensus/tx_verify.h>
+#include <consensus/validation.h>
+#include <index/txindex.h>
+#include <script/script.h>
+#include <test/test_bitcoin.h>
+#include <util/memory.h>
+
+#include <memory>
+
+#include <boost/test/unit_test.hpp>
+
+static CMutableTransaction createCoinbase() {
+	CMutableTransaction coinbaseTx;
+
+	coinbaseTx.nVersion = 1;
+	coinbaseTx.vin.resize(1);
+	coinbaseTx.vout.resize(1);
+	coinbaseTx.vin[0].scriptSig = CScript() << OP_11 << OP_EQUAL;
+	coinbaseTx.vin[0].prevout.SetNull();
+	coinbaseTx.vout[0].nValue = 50000;
+	coinbaseTx.vout[0].scriptPubKey = WDMO_SCRIPT;
+
+	return coinbaseTx;
+}
+
+static CMutableTransaction createLicenseTransaction(const uint256 parentHash) {
+	CMutableTransaction tx;
+	tx.vin.resize(1);
+	tx.vin[0].scriptSig = CScript() << OP_11;
+	tx.vin[0].prevout.hash = parentHash;
+	tx.vin[0].prevout.n = 0;
+	tx.vout.resize(2);
+	tx.vout[0].scriptPubKey = CScript();
+	tx.vout[0].nValue = 49000;
+	tx.vout[1].scriptPubKey = CScript() << OP_RETURN << std::vector<unsigned char>{3, 76, 84, 120, 20, 169, 20, 230, 53, 247, 106, 15, 189, 177, 112, 86, 88, 161, 227, 176, 86, 115, 120, 7, 74, 0, 5};
+	tx.vout[1].nValue = 0;
+
+	return tx;
+}
+
+struct DdmsSetup : public TestChain100Setup {
+	DdmsSetup()
+		: TestChain100Setup() {
+		minerLicenses = MinerLicenses{};
+	}
+	~DdmsSetup() = default;
+};
+
+BOOST_FIXTURE_TEST_SUITE(ddms_tests, DdmsSetup)
+
+BOOST_AUTO_TEST_CASE(shouldIsLicenseTxHeaderReturnTrueWhenProcessingLTxScriptPubKey)
+{
+	CScript ltxScriptPubKey = CScript() << OP_RETURN << std::vector<unsigned char>{3, 76, 84, 120, 20, 169, 20, 230, 53, 247, 106, 15, 189, 177, 112, 86, 88, 161, 227, 176, 86, 115, 120, 7, 74, 0, 5};
+	BOOST_CHECK(IsLicenseTxHeader(ltxScriptPubKey));
+}
+
+BOOST_AUTO_TEST_CASE(shouldIsLicenseTxHeaderReturnFalseWhenNotProcessingLTxScriptPubKey)
+{
+	CScript notLtxScriptPubKey = CScript() << OP_RETURN << std::vector<unsigned char>{2, 31, 72, 20, 169, 20, 230, 53, 247, 106, 15, 189, 177, 112, 86, 88, 161, 227, 176, 86, 115, 120, 7, 74, 0, 5};
+	BOOST_CHECK(!IsLicenseTxHeader(notLtxScriptPubKey));
+}
+
+BOOST_AUTO_TEST_CASE(shouldIsLicenseTxReturnFalseWhenTxNullOrCoinbase)
+{
+	CMutableTransaction coinbaseTx, nullTx;
+	coinbaseTx = createCoinbase();
+
+	BOOST_CHECK(!IsLicenseTx(CTransaction(nullTx)));
+	BOOST_CHECK(!IsLicenseTx(CTransaction(coinbaseTx)));
+}
+
+BOOST_AUTO_TEST_CASE(shouldIsLicenseTxReturnFalseWhenTxWasNotSentByWDMO)
+{
+	g_txindex = MakeUnique<TxIndex>(1 << 20, true);
+	g_txindex->Start();
+
+	CScript fakeWdmoScript = WDMO_SCRIPT; --fakeWdmoScript[1];
```

Comment:
`--fakeWdmoScript[1]` will cause failure in the script syntax itself, since this byte specifies the length of script hash pushed to the stack. To keep it cleaner, modify the values in the hash itself, e.g. byte no 10.

Reply:
Done